file-validator.py

```diff
@@ -0,0 +1,239 @@
+    historicalComparison: str
+    
+    
+def is_yaml_file(file_path):
+    _, file_extension = os.path.splitext(file_path)
+    return file_extension.lower() == '.yaml'
+
+def manualMetaDataImport():
+    #ask name for the folder
+    folderName = input("Folder Name: ")
+    os.makedirs(folderName)
+    #build index
+    nbPic=buildIndex(folderName)
+    #for each picture in field
+    zoomlevel = input("Zoom level for this index: ")
+    seedNumber = input("Number of seed for this index: ")
+    for x in range(nbPic):
+        buildPicture(folderName,zoomlevel,seedNumber)
+
+def buildIndex(output:str):
+    # Ask for the url in the console
+
+    # # Retrieve some client info from the URL
+
+    # Ask Data (I dont have access to the data url atm)
+    clientEmail=input("clientEmail: ")
+    clientExpertise= input("Expertise: ")
+    clientData=ClientData(clientEmail=clientEmail,clientExpertise=clientExpertise)
+    nb=input("numberOfImages: ")
+    imageData=ImageDataindex(numberOfImages=nb)
+    seedID=input("SeedID: ")
+    family = input("Family: ")
+    genus = input("Genus: ")
+    species= input("Species: ")
+    seedData=SeedData(seedID=seedID,seedFamily=family,seedGenus=genus,seedSpecies=species)
+    
+    index = Index(clientData=clientData,imageData=imageData,seedData=seedData)
+    name = input("File created, name: ")
+    
+    #createJsonIndex(index=index,name=name,output=output)
+    indexJson=index.dict()
+    sysData=IndexProcessing(openIndexSystem())
+    indexJson.update(sysData)
+    print(indexJson)
+    createJsonIndex(index=indexJson,name=name,output=output)
+    return nb
+
+
+def buildPicture(output:str,number:int,level):
+    # Ask for the url in the console
+
+    # # Retrieve some client info from the URL
+
+    # Ask Data (I dont have access to the data url atm)
+    desc= ""
+    nb=number
+    zoom=level
+    userData = UserData(description=desc,numberOfSeeds=nb,zoom=zoom)
+    
+    pic = Picture(UserData=userData)
+    name = input("File created, name: ")
+    
+    createJsonPicture(pic=pic,name=name,output=output)
+
+#Create Json form Picture data model
+def createJsonPicture(pic:Picture,name:str,output:str):
+    data = pic.dict()
+    filePath = f'{output}/{name}.json'
+    with open(filePath,'w') as json_file:
+        json.dump(data,json_file,indent=2)
+    
+# Create Json from Index data model
+def createJsonIndex(index,name: str,output:str):
+    #data=index.dict()
+    data = PIndex(**index)
+    filePath = f'{output}/{name}.json'
+    with open(filePath,'w') as json_file:
+        json.dump(index,json_file,indent=2)
+    
+    
+    
+def openIndex(path):
+    if is_yaml_file(path):
+        # Load YAML template
+        with open(path, 'r') as file:
+            yaml_data = yaml.safe_load(file)
+    try: # Validate and parse YAML data using Pydantic model
+        data = Index(**yaml_data)
+    except Exception as error: # Errtor raised by Pydantic Validator
+        print(error)
```

Comment:
should the program really continues without failing? typically should re-raise your custom exception to ensure the caller handles failure here.